src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ReportDocHelper.java

```diff
@@ -55,62 +61,76 @@ public static StringBuilder buildReportDocumentContent(ListingData listingData,
+            case CLAIMS_ACCEPTED_REPORT:
                 sb.append(ListingHelper.getListingDate(listingData));
-                sb.append(getHearingsToJudgmentsReport(listingData));
-            } catch (JsonProcessingException e) {
-                throw new ReportException("Unable to create report data", e);
-            }
-        } else {
-            sb.append(ListingHelper.getListingDate(listingData));
-
-            if (listingData.getLocalReportsDetailHdr() != null) {
-                sb.append(REPORT_OFFICE).append(
-                        nullCheck(listingData.getLocalReportsDetailHdr().getReportOffice())).append(NEW_LINE);
-            } else if (CollectionUtils.isNotEmpty(listingData.getLocalReportsSummary())) {
-                sb.append(REPORT_OFFICE).append(
-                        nullCheck(listingData.getLocalReportsSummary().get(0)
-                                .getValue().getReportOffice())).append(NEW_LINE);
-            }
-            switch (listingData.getReportType()) {
-                case CLAIMS_ACCEPTED_REPORT:
-                    sb.append(getCasesAcceptedReport(listingData));
-                    break;
-                case LIVE_CASELOAD_REPORT:
-                    sb.append(getLiveCaseLoadReport(listingData));
-                    break;
-                case CASES_COMPLETED_REPORT:
-                    sb.append(getCasesCompletedReport(listingData));
-                    break;
-                case TIME_TO_FIRST_HEARING_REPORT:
-                    sb.append(getTimeToFirstHearingReport(listingData));
-                    break;
-                case CASE_SOURCE_LOCAL_REPORT:
-                    sb.append(getCaseSourceLocalReport(listingData));
-                    break;
-                case SERVING_CLAIMS_REPORT:
-                    sb.append(getServedClaimsReport(listingData));
-                    break;
-                case HEARINGS_BY_HEARING_TYPE_REPORT:
-                    sb.append(getHearingsByHearingTypeReport(listingData));
-                    break;
-                default:
-                    throw new IllegalStateException("Report type - Unexpected value: " + listingData.getReportType());
-            }
+                addReportOffice(listingData, sb);
+                sb.append(getCasesAcceptedReport(listingData));
+                break;
+            case LIVE_CASELOAD_REPORT:
+                sb.append(ListingHelper.getListingDate(listingData));
+                addReportOffice(listingData, sb);
+                sb.append(getLiveCaseLoadReport(listingData));
+                break;
+            case CASES_COMPLETED_REPORT:
+                sb.append(ListingHelper.getListingDate(listingData));
+                addReportOffice(listingData, sb);
+                sb.append(getCasesCompletedReport(listingData));
+                break;
+            case TIME_TO_FIRST_HEARING_REPORT:
+                sb.append(ListingHelper.getListingDate(listingData));
```

Comment:
Why is addReportOffice being called in each case? May be it can be called after case statement? Same with sb.append(ListingHelper.getListingDate(listingData));

Reply:
or may be before case statement

Reply:
there's a couple of reports that don't use the methods, so i can't just blanket add it to everything...

Reply:
Have approved it but there are just two reports not assigning report office and date both, and one report only which doesn't assign reportOffice. Also LISTING_DATA_STATE_EXCEPTION should be handled/thrown from it's own method rather than from switch case statements as it is done in case awaiting judgmement and hearings by judgement type reports.